src/main/java/org/tugraz/sysds/runtime/matrix/data/FrameBlock.java

```diff
@@ -1600,4 +1601,18 @@ public void setMvValue(String mvVal) {
 			_mvValue = mvVal;
 		}
 	}
-}
+
+	public FrameBlock getSchemaTypeOf()
+	{
+		ValueType[] schemaStrings = new ValueType[this.getNumColumns()];
+		for(int i=0; i<schemaStrings.length; i++)
+			schemaStrings[i] = ValueType.STRING;
+
+		String[][] st = new String[1][this.getNumColumns()];
```

Comment:
Avoid the creation of this two dimensional array - you can append a single string row to an empty frame.